lib/cucumber-api/steps.rb

```diff
@@ -153,13 +171,24 @@
   end
 end
 
+Then(/^the JSON response should have "([^"]*)" of type \
+(numeric|string|boolean|numeric_string) and value "([^"]*)"$/) do |json_path, type, value|
+  @response.get_as_type_and_check_value json_path, type, resolve(value)
+end
+
 # Bind grabbed values into placeholders in given URL
 # Ex: http://example.com?id={id} with {id => 1} becomes http://example.com?id=1
 # @param url [String] parameterized URL with placeholders
 # @return [String] binded URL or original URL if no placeholders
 def resolve url
-  unless @grabbed.nil?
-    @grabbed.each { |key, value| url = url.gsub /\{#{key}\}/, %/#{value}/ }
+  url.gsub!(/\{([a-zA-Z0-9_]+)\}/) do |s|
+    s.gsub!(/[\{\}]/, '')
+    if instance_variable_defined?("@#{s}")
+      instance_variable_get("@#{s}")
+    else
+      raise 'Undefined instance variable: ' + s
```

Comment:
Maybe adding a message along the line of `Have you forgotten to grab {something}?` would be helpful?